### ISSUE TITLE
[Enum] Promote constant visibility as a replacement of Enum::getIgnoredConstantNames()

### ISSUE THREAD

**Issue by Majkl578** [consistence/consistence#24]
This is just a quick idea, but would it make sense to deprecate getIgnoredConstantNames() in favor of native constant visibility? Example:

```php
final class SomeEnum extends Consistence\Enum\Enum
{
    public const FOO = 1;
    public const BAR = 2;

    private const MAP = [
        self::FOO => 'foo',
        self::BAR => 'bar',
    ];

    public static function getSomeName() : string
    {
        return self::MAP[$this->getValue()];
    }

    public static function getIgnoredConstantNames() : array
    {
        return ['MAP'];
    }
}

var_dump(SomeEnum::getAvailableValues()); // ['FOO' => 1, 'BAR'  => 2]
```

What I propose is deprecating & removing `getIgnoredConstantNames()` and relying solely on visibility. So:
```php

final class SomeEnum extends Consistence\Enum\Enum
{
    public const FOO = 1;
    public const BAR = 2;

    private const MAP = [
        self::FOO => 'foo',
        self::BAR => 'bar',
    ];

    public static function getSomeName() : string
    {
        return self::MAP[$this->getValue()];
    }

    // no getIgnoredConstantNames()
}

var_dump(SomeEnum::getAvailableValues()); // ['FOO' => 1, 'BAR'  => 2], private MAP is ignored automatically
```

**Comment by VasekPurchart** [comment]
Thanks for the suggestion, I thought about something similar. At first I thought about keeping `getIgnoredConstantNames()` and implementing ignoring non-public constant in that, but you are probably right and we will be better without it (because when you override `getAvailableValues()` it might never be called.

Sorry for late reply, was skiing outside of the country :)

**Comment by VasekPurchart** [comment]
Implemented in #38.
